### PR TITLE
Migrate user notifications view to bootstrap

### PR DESCRIPTION
**Pull request by krauselukas**
The notifications are part of the user infos which gets
migrated to bootstrap

screenshot-2018-11-27 notifications - open build service

### CODE REVIEW

**Comment by krauselukas** [comment]
@hennevogel i could arrange the check boxes in a row, like in the old view, besides that i haven't changed much. I thought it more readable this way, even though i need more space

**Comment by krauselukas** [comment]
@hennevogel Changed it
screenshot-2018-11-27 notifications - open build service 1

File: src/api/app/views/webui2/webui/subscriptions/_subscriptions_form.html.haml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+:ruby
+  i = 0
+
+- @subscriptions_form.subscriptions_by_event.each do |event_class, subscriptions|
+  %li.list-group-item
+    %h5= event_class.description
+    %p= Event::Base::EXPLANATION_FOR_NOTIFICATIONS[event_class.to_s]
+    - subscriptions.each do |subscription|
```
[the remark this answers is not in the document]
Comment (coolo) [reply src/api/app/views/webui2/webui/subscriptions/_subscriptions_form.html.haml]: careful - there is subscriptions_by_event.each loop around it.

So before you go fancy, make sure you actually have tests for this view.

[the remark this answers is not in the document]
Comment (krauselukas) [reply src/api/app/views/webui2/webui/subscriptions/_subscriptions_form.html.haml]: @coolo @DavidKang yes i tried it, breaks things. But i will have closer look tomorrow.

File: src/api/app/views/webui2/webui/users/subscriptions/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - if @user.rss_token
+        To access your RSS feeds, use the following url:
+        = link_to(user_rss_notifications_url(token: @user.rss_token.string, format: 'rss'),
+        user_rss_notifications_url(token: @user.rss_token.string, format: 'rss'), target: '_blank')
+      - else
+        No feed url exists for your notifications.
+      .card-body
+        = submit_tag "#{@user.rss_token ? 'Regenerate Url' : 'Generate Url'}", class: 'btn btn-primary'
+
+
```
[the remark this answers is not in the document]
Comment (krauselukas) [reply src/api/app/views/webui2/webui/users/subscriptions/index.html.haml]: Done

File: src/api/app/views/webui2/webui/users/subscriptions/index.js.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+$('#subscriptions-form').replaceWith('<%= escape_javascript(render(partial: 'webui/subscriptions/subscriptions_form')) %>');
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/users/subscriptions/index.js.erb]: We don't use ERB for new files.

Comment (coolo) [reply src/api/app/views/webui2/webui/users/subscriptions/index.js.erb]: but?

Comment (dmarcoux) [reply src/api/app/views/webui2/webui/users/subscriptions/index.js.erb]: We use HAML now.

Comment (coolo) [reply src/api/app/views/webui2/webui/users/subscriptions/index.js.erb]: but this is not html, but javascript

Comment (dmarcoux) [reply src/api/app/views/webui2/webui/users/subscriptions/index.js.erb]: I think this should be inline in the subscriptions/index view with `javascript: [..]`. Then this file should be deleted. Sorry for the confusion @krauselukas, I should have been clearer from the start.

Comment (bgeuken) [reply src/api/app/views/webui2/webui/users/subscriptions/index.js.erb]: This view is probably rendered when the `update` action is called. If you move it to the html views, you might have to update how we handle the ajax requests (and their response).

And changing that might also affect the old views. I would postpone this until after the migration.

Comment (krauselukas) [reply src/api/app/views/webui2/webui/users/subscriptions/index.js.erb]: @dmarcoux would you be fine with postponing it?

Comment (dmarcoux) [reply src/api/app/views/webui2/webui/users/subscriptions/index.js.erb]: Yes, no worries.

**Comment by dmarcoux** [comment]
Hakiri is stuck again...

**Pull request merged**